libavcodec/qsvenc.c

```diff
@@ -1648,6 +1655,59 @@ static void print_interlace_msg(AVCodecContext *avctx, QSVEncContext *q)
     }
 }
 
+static int update_qp(AVCodecContext *avctx, QSVEncContext *q,
+                     const AVFrame *frame)
+{
+    int updated = 0, qp = 0;
+    AVDictionaryEntry *entry = av_dict_get(frame->metadata, "qsv_config_qp",
+                                           NULL, 0);
+    if (entry) {
+        qp = atoi(entry->value);
+        av_log(avctx, AV_LOG_DEBUG, "Configure qp: %d\n",qp);
+        if (qp < 1 || qp > 51) {
```

Comment:
Need to consider jpeg/vp9/av1 here? Their max qp may not 51.

Reply:
I see qsvenc is using 51 so I use 51 as well. [https://github.com/intel-media-ci/ffmpeg/blob/e3580f60775c897c3b13b178c57ab191ecc4a031/libavcodec/qsvenc.c#L816](https://github.com/intel-media-ci/ffmpeg/blob/e3580f60775c897c3b13b178c57ab191ecc4a031/libavcodec/qsvenc.c#L816 ). Yes it is an issue for vp9. Maybe we need to send another patch to fix qp problem for qsvenc.

Reply:
done, thanks.

Reply:
Hi Fei. The range listed in doc is 0~51. I discussed with Haihao and I decide to tell MediaSDK team to update doc. I am going to submit this patch using value 51.  After the doc is updated I will submit another patch to fix qp range issue in qsvenc.